### PR TITLE
[tp-25]-protected route

### LINKED ISSUE #25: protected route
read docs

### PR DESCRIPTION
**Pull request by altNameForStudying**
* Uses Outlet as Route Wrapper
* Temporary logic for handling userState implemented (basic method with `logged in` property set to `false`)
* Refactored routes; now only contain paths
* Views.js for Routes definitions
* Updated ProfileTabs so it uses Views.js

https://user-images.githubusercontent.com/86930816/150615674-96a47b78-2c88-4c21-8a17-6dda946c2b82.mp4

https://user-images.githubusercontent.com/86930816/150615676-f9eaebda-a46e-4dd8-8ce7-84dbc53207bf.mp4

### CODE REVIEW

File: src/Views.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import Menus from './pages/menus/Menus';
+import Recipes from './pages/recipes/Recipes';
+import { ROUTES_PATHS } from './routes';
+
+const Views = () => {
+    return (
+        <Routes>
+            <Route path={ROUTES_PATHS.MAIN_PAGE} element={<MainPage />} key={'main-page'} />
+            <Route element={<ProtectedRoutes />}>
+                <Route
```
Comment (pdybowski) [inline src/Views.js]: cant you load routes dynamically like it was before?

File: src/components/userProfile/profileTabs/ProfileTabs.jsx [hunk 2]
```diff
@@ -37,13 +38,9 @@ function ProfileTabs({ tab = PROFILE_TABS.PROFILE_RECIPES }) {
 }
 
 function getRouteFromKey(key) {
-    return routes.find((route) => route.key === key).path;
+    console.log(Object.keys(Views).find((route) => route.key === key).path);
+    // return routes.find((route) => route.key === key).path;
```
Comment (pdybowski) [inline src/components/userProfile/profileTabs/ProfileTabs.jsx]: you can removethese two lines

File: src/routes.js [hunk 3]
```diff
@@ -19,42 +14,8 @@ export const ROUTES_PATHS = {
     USER_FAVORITES: '/user-profile/favourites',
 };
 
-export const routes = [
```
Comment (pdybowski) [inline src/routes.js:22]: can you leave it here and just map these roues inside view file?

File: src/routes.js [hunk 4]
```diff
@@ -19,42 +14,8 @@ export const ROUTES_PATHS = {
...
-        element: <Recipes />,
-        key: 'recipes',
-    },
-    {
-        path: ROUTES_PATHS.MENUS,
-        element: <Menus />,
-        key: 'menus',
-    },
-];
+export const PROFILE_TABS = {
```
Comment (pdybowski) [inline src/routes.js:21]: why is here? is it realetd to routes or to tabs?

**Review by pdybowski — CHANGES_REQUESTED** [review]
please fix

File: src/routes.js (hunk 4 above)
Comment (altNameForStudying) [reply src/routes.js:21]: To both routes and tabs as you can navigate using these urls directly to user view. 
Due to that i figured it's better to keep all accessible routes in the same place

**Review by pdybowski — APPROVED** [review]

**Pull request merged**